carta/image.py

```diff
@@ -605,6 +605,189 @@ def hide_contours(self):
         """Hide the contours."""
         self.set_contours_visible(False)
 
+    # VECTOR OVERLAY
+
+    @validate(NoneOr(Constant(VectorOverlaySource)), NoneOr(Constant(VectorOverlaySource)), NoneOr(Number()), Boolean(), NoneOr(Number()), NoneOr(Number()), NoneOr(Number()))
+    def configure_vector_overlay(self, angular_source=None, intensity_source=None, pixel_averaging=4, fractional_intensity=False, threshold=None, q_error=None, u_error=None):
+        """Configure vector overlay.
+
+        Parameters
+        ----------
+        angular_source : {0}
+            The angular source. If the image is with Stoke information, ``Computed PA`` is set by default; If the image is without Stoke information, ``Current image`` is set.
+        intensity_source : {1}
+            The intensity source. If the image is with Stoke information, ``Computed PI`` is set by default; If the image is without Stoke information, ``Current image`` is set.
+        pixel_averaging : {2}
+            The pixel averaging width in pixel. Set to ``None`` to disable pixel averaging.
+        fractional_intensity : {3}
+            Enable fractional polarization intensity. By default the absolute polarization intensity is used.
+        threshold : {4}
+            The threshold in Jy/pixels. By default is 4.
+        q_error : {5}
+            The Stokes Q error in Jy/beam. Set both this and ``u_error`` to enable debiasing. Debiasing is disabled by default.
+        u_error : {6}
+            The Stokes U error in Jy/beam. Set both this and ``q_error`` to enable debiasing. Debiasing is disabled by default.
+        """
+        pixel_averaging_enabled = pixel_averaging is not None
+        threshold_enabled = threshold is not None
+        debiasing = q_error is not None and u_error is not None
+        if not debiasing and (q_error is not None or u_error is not None):
+            logger.warning("The Stokes Q and the Stokes U must both be set to enable debiasing.")
+        if angular_source is None:
+            angular_source = "None"
+        if intensity_source is None:
+            intensity_source = "None"
+        if threshold is None:
+            threshold = "None"
+        if q_error is None:
+            q_error = "None"
+        if u_error is None:
+            u_error = "None"
+        self.call_action("vectorOverlayConfig.setVectorOverlayConfiguration", angular_source, intensity_source, pixel_averaging_enabled, pixel_averaging, fractional_intensity, threshold_enabled, threshold, debiasing, q_error, u_error)
+
+    @validate(Number(), NoneOr(Number()), NoneOr(Number()), Number(), Number(), Number())
+    def set_vector_overlay_style(self, thickness=1, intensity_min=None, intensity_max=None, length_min=0, length_max=20, rotation_offset=0):
```

Comment:
For consistency with the approach of ignoring options which are not explicitly provided (and keeping the currently set frontend values), all these hardcoded defaults should be removed, and all these parameters should default to `None`.

You don't need to use any macros in here, but you do need to add guards to all the individual action calls, and only call the actions if the parameters are not `None`. (The intensity range logic is fine; the action won't be called if both values are `None`, which is correct.)